### ISSUE TITLE
Allow @Preview functions to not take Modifier arguments

### ISSUE THREAD

**Issue by inktomi** [twitter/compose-rules#41]
In general, `detekt.ModifierMissing` is for sure the right approach. Where it gets a little less useful though is on `@Preview` methods. In general, at least in my projects, these are rarely (if ever) used from inside another `@Composable`. 

Is there an appetite for a PR adjusting this rule to not match against methods that are also annotated with `@Preview`? 

For instance, I'd suggest that `ModifierMissing` not be flagged against this method: 

```kotlin
@Composable
@Preview(name = "Container with Single Product", uiMode = Configuration.UI_MODE_NIGHT_NO)
@Preview(name = "Container with Single Product", uiMode = Configuration.UI_MODE_NIGHT_YES)
fun OverviewWithSwitcherContainerWithSingleProductPreview() {
 //
} 
```

**Comment by mrmans0n** [comment]
Yep, this is a good idea, I'll implement it in the `ModifierMissing` rule.

It also brings something to mind. The modifier rule wouldn't be applied (on purpose) if the composable is not public (e.g. internal or private), and I don't know of a reason why a preview would have public visibility. Maybe we should also add a rule to check for that: flag `@Preview` composables that have public visibility.
